sources/engine/Xenko.Voxels/Voxels/Voxelization/Attributes/VoxelAttributeBase.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xenko.Core;
+using Xenko.Core.Annotations;
+using Xenko.Shaders;
+using Xenko.Rendering.Materials;
+using Xenko.Core.Mathematics;
+
+namespace Xenko.Rendering.Voxels
+{
+    public abstract class VoxelAttributeBase
```

Comment:
An abstract class forces any class deriving from it to implement any functions and properties marked as abstract;
you can merge ``IVoxelAttribute`` with this class now that it's abstract, you just have to mark all of the properties and functions that where inside of ``IVoxelAttribute`` as abstract.

You can then remove ``IVoxelAttribute`` and replace any reference to that type with ``VoxelAttributeBase``.